packages/azure-kusto-ingest/src/fileDescriptor.browser.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { CompressionType, getSourceId } from "./descriptors";
+
+export class FileDescriptor {
+    size: number | null;
+    zipped: boolean;
+    compressionType: CompressionType;
+    cleanupTmp?: () => Promise<void>;
+
+    constructor(
+        readonly file: Blob,
+        readonly sourceId: string | null = null,
+        size: number | null = null,
+        compressionType: CompressionType = CompressionType.None,
+        readonly extension?: string,
+        readonly name?: string
+    ) {
+        this.sourceId = getSourceId(sourceId);
+        this.compressionType = compressionType;
+        this.size = size || file.size;
+        this.zipped = compressionType !== CompressionType.None || this.extension === ".gz" || this.extension === ".zip";
+    }
+
+    // Not used
+    async _gzip(): Promise<string> {
```

Comment:
This should work, probably

Reply:
what ?

Reply:
I mean, we should gzip the file even in browsers, not leave it empty.

Reply:
Where? how?

Reply:
Did you?
async _gzip(): Promise<string> {
return await this.file.text();
}
???

Reply:
Why does it exist at all? it's not in the base

Reply:
will delete then